lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -1187,6 +1190,58 @@ void MayaUsdProxyShapeBase::_OnStageContentsChanged(const UsdNotice::StageConten
 void MayaUsdProxyShapeBase::_OnStageObjectsChanged(const UsdNotice::ObjectsChanged& notice)
 {
     ProxyAccessor::stageChanged(_usdAccessor, thisMObject(), notice);
+
+    // Recompute the extents of any UsdGeomBoundable that has authored extents
+    const auto& stage = notice.GetStage();
+    if (stage != getUsdStage()) {
+        TF_CODING_ERROR("We shouldn't be receiving notification for other stages than one "
+                        "returned by stage provider");
+        return;
+    }
+
+    UsdEditContext editContext(stage, stage->GetSessionLayer());
```

Comment:
Here is a bit of a problem. The session layer is not serialized together with the root layer. 

I would assume that this notification will happen right after the authoring operation and for that, we already have the right edit target set. What's the reason for changing the edit target?

Reply:
Yes, this line introduces bugs. I will remove it.